ivy_tests/test_ivy/test_frontends/test_torch/test_nn/test_functional/test_pooling_functions.py

```diff
@@ -215,23 +216,13 @@ def test_torch_max_pool1d(
     on_device,
 ):
     input_dtype, x, kernel_size, stride, padding = dtype_x_k_s
-
-    # Torch ground truth func expects input to be consistent
-    # with a channels first format i.e. NCW
-    x[0] = x[0].reshape((x[0].shape[0], x[0].shape[-1], x[0].shape[1]))
     x_shape = [x[0].shape[2]]
-
     # Torch ground truth func also takes padding input as an integer
     # or a tuple of integers, not a string
-    padding = tuple(
-        [
-            ivy.functional.layers._handle_padding(
-                x_shape[i], stride[0], kernel_size[i], padding
-            )
-            for i in range(len(x_shape))
-        ]
-    )
-
+    if padding == "SAME":
+        padding = calculate_same_padding(kernel_size, stride, x_shape)
+    elif padding == "VALID":
+        padding = (0,)
```

Comment:
If we keep `only_explicit_padding=True` we can avoid handling `"SAME"` and `"VALID"` altogether. I think that would be best.